src/test/java/teammates/sqlui/webapi/GetCoursesActionTest.java

```diff
@@ -0,0 +1,190 @@
+        assertEquals(stubCourseList.get(0).getId(), data.getCourses().get(0).getCourseId());
+        assertEquals(stubInstructor.getPrivileges().getCourseLevelPrivileges(), data.getCourses().get(0).getPrivileges());
+    }
+
+    @Test
+    void testExecute_withInstructorAndInvalidCourseStatus_throwsException() {
+        loginAsInstructor(stubInstructor.getGoogleId());
+        String[] params = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+                Const.ParamsNames.COURSE_STATUS, "invalid",
+        };
+        verifyHttpParameterFailure(params);
+
+        String[] params2 = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+                Const.ParamsNames.COURSE_STATUS, null,
+        };
+
+        verifyHttpParameterFailure(params2);
+    }
+
+    @Test
+    void testExecute_withStudentEntityType_success() {
+        loginAsStudent("student");
+        when(mockLogic.getCoursesForStudentAccount("student")).thenReturn(stubCourseList);
+        String[] params = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+        GetCoursesAction action = getAction(params);
+        JsonResult result = action.execute();
+        CoursesData data = (CoursesData) result.getOutput();
+        assertEquals(1, data.getCourses().size());
+        assertEquals(stubCourse.getId(), data.getCourses().get(0).getCourseId());
+        assertEquals(0, data.getCourses().get(0).getDeletionTimestamp());
+    }
+
+    @Test
+    void testExecute_withInvalidEntityType_throwsException() {
+        loginAsStudent("student");
+        String[] params = {
+                Const.ParamsNames.ENTITY_TYPE, "invalid",
+        };
+        verifyHttpParameterFailure(params);
+
+        String[] params2 = {
+                Const.ParamsNames.ENTITY_TYPE, null,
+        };
+        verifyHttpParameterFailure(params2);
+    }
+
+    @Test
+    void testSpecificAccessControl_instructor_success() {
+        loginAsInstructor(stubInstructor.getGoogleId());
+        String[] params = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+                Const.ParamsNames.COURSE_STATUS, Const.CourseStatus.ACTIVE,
+        };
+        verifyCanAccess(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_student_success() {
+        loginAsStudent("student");
+        String[] params = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+        verifyCanAccess(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_admin_cannotAccess() {
+        loginAsAdmin();
+        String[] params = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.ADMIN,
+                Const.ParamsNames.COURSE_STATUS, Const.CourseStatus.ACTIVE,
+        };
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_instructorAndNotLoggedIn_cannotAccess() {
```

Comment:
I think we can test for student params and not logged in this same method. Something like this:

```
String[] studentParams = new String[] { Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT, };
String[] instructorParams = new String[] { Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR, };

logoutUser();
verifyCannotAccess(studentParams);
verifyCannotAccess(instructorParams);
```